doc/Language/rb-nutshell.pod

```diff
@@ -0,0 +1,944 @@
+    puts months[8..11].join(',') # Ruby
+    say @months[8..11].join(',') # Perl 6
+=end item
+
+Also note that the subscripting brackets are a normal postcircumfix operator
+rather than a special syntactic form, and thus L<checking for existence of
+elements|#exists> and L<unsetting elements|#delete> is done with adverbs.
+
+=head3 {} Hash indexing/slicing
+
+Ruby uses square brackets to access values for both Arrays and Hashes. Perl 6
+uses curley braces for hashes instead. The angle-brackets version is available
+which always autoquotes its contents (strings without quotes):
+
+Adverbs can be used to control the type of slice.
+
+=begin item
+I<Indexing>
+
+    puts calories["apple"]  # Ruby
+    say %calories{"apple"}; # Perl 6 - % instead of $
+
+    puts calories["apple"]  # Ruby
+    puts calories[:apple]   # Ruby, symbols for keys are common
+    say %calories<apple>;   # Perl 6 - angle brackets instead of single-quotes
+    say %calories«$key»;    # Perl 6 - double angles interpolate as double-quotes
+=end item
+
+=begin item
+I<Value-slicing>
+
+    puts calories.values_at('pear', 'plum').join(',') # Ruby
+    puts calories.values_at(%w(pear plum)).join(',')  # Ruby, pretty?
+    say %calories{'pear', 'plum'}.join(',');          # Perl 6
+    say %calories<pear plum>.join(',');               # Perl 6 (prettier)
+    my $keys = 'pear plum';
+    say %calories«$keys».join(','); # Perl 6, interpolated split
+=end item
+
+=begin item
+I<Key/value-slicing>
+
+    say calories.slice('pear', 'plum').join(','); # Ruby, with ActiveRecord
+    say %calories{'pear', 'plum'}:kv.join(',');   # Perl 6 - use :kv adverb
+    say %calories<pear plum>:kv.join(',');        # Perl 6 (prettier version)
+=end item
+
+Also note that the subscripting curly braces are a postcircumfix operator
+rather than a special syntactic form, and thus L<checking for existence of
+keys|#exists> and L<removing keys|#delete> is done with adverbs.
+
+=head3 C<:> Symbols
+
+Perl 6 generally uses strings in the places where Ruby uses symbols. A primary
+example of this is in hash keys.
+
+    address[:joe][:street] # Typical Ruby nested hash with symbol keys
+    %address<joe><street>  # Typical Perl 6 nested hash with string keys
+
+Just to be confusing, Perl 6 has I<colon-pair> syntax, which can sometimes look
+like Ruby symbols.
+
+    :age            # Ruby symbol
+    :age            # Perl 6 pair with implicit True value
+    :age(True)      # Perl 6 pair with explicit True value
+    age => True     # Perl 6 pair using arrow notation
+    "age" => True   # Perl 6 pair using arrow notation and explicit quotes
+
+You could probably get away with using a colon-pair without an explicit value
+and pretend that it is a Ruby symbol a lot of the time, but it is probably a
+bad idea.
+
+=head3 Variable Declaration and Scope
+
+For local variables, Ruby uses implicit variable declaration upon assignment
+and limited to the current block. Perl 6 uses explicit scope indicators, and
```

Comment:
"block" in Ruby is looser than in P6. In ruby a do/end block has a scope, but the content of an if or while loop do not. In P6 even if and while loop contents are scopes. Maybe this doesn't matter here though.